cmd/archeio/main.go

```diff
@@ -70,31 +76,33 @@ func main() {
 	}
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	// right now we just need to serve a redirect, but all
-	// valid requests should be at /v2/ or /v1/, so we leave this check
-	// in the future we will selectively redirect clients to different copies
-	path := r.URL.Path
-	switch {
-	case strings.HasPrefix(path, "/v2/"):
-		doV2(w, r)
-	case strings.HasPrefix(path, "/v1/"):
-		doV1(w, r)
-	default:
-		klog.V(2).InfoS("unknown request", "path", path)
-		http.NotFound(w, r)
-	}
+func makeHandler(upstreamRegistry string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// right now we just need to serve a redirect, but all
+		// valid requests should be at /v2/ or /v1/, so we leave this check
+		// in the future we will selectively redirect clients to different copies
+		path := r.URL.Path
+		switch {
+		case strings.HasPrefix(path, "/v2/"):
+			doV2(w, r, upstreamRegistry)
+		case strings.HasPrefix(path, "/v1/"):
+			doV1(w, r, upstreamRegistry)
```

Comment:
Do we still need this ? GCR stopped to serve `v1` API. 

```she curl -I https://k8s.gcr.io/v1/pause/tags/list
HTTP/2 404
date: Mon, 21 Mar 2022 21:46:57 GMT
content-type: text/html; charset=UTF-8
server: Docker Registry
content-length: 1579
x-xss-protection: 0
x-frame-options: SAMEORIGIN
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
```

Reply:
let's do that cleanup in a different PR, v1 support here is purely a refactor